avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -274,14 +274,23 @@ public Object fromConnectData(Schema schema, Object value) {
    * @param logicalValue the Connect data to convert, which may be a value for a logical type
    * @param requireContainer if true, wrap primitives, maps, and arrays in a NonRecordContainer
    *                         before returning them
+   * @param requireSchemalessContainerNull if true, use a container representation of null because
+   *                                       this is part of struct/array/map and we cannot
+   *                                       represent nulls as true null because Anything cannot
+   *                                       be a union type; otherwise, this is a top-level value
+   *                                       and can return null
    * @return the converted data
    */
   private static Object fromConnectData(Schema schema, org.apache.avro.Schema avroSchema,
-                                       Object logicalValue, boolean requireContainer) {
+                                       Object logicalValue, boolean requireContainer, boolean
+                                            requireSchemalessContainerNull) {
```

Comment:
@gwenshap The parameters is kind of confusing. The problem is that it's not based on the schema type, it's based on the *parent* schema type. So when you are translating *any* type, but it is an element in a map/array, you need to encode as an Anything struct with no entries to indicate null. This method gets called once for the top level object (e.g. the array) and once for each element. If the top-level array is null, we want to actually make it a null (e.g. relevant for deletion in compacted topics). But for each member element, we need to encode it as the struct form since Anything cannot be a union[null, everything else].

I'm actually beginning to think I should refactor into separate `fromConnectDataWithSchema` and `fromConnectDataSchemaless` to break this up a bit, and maybe make some of the parameters a bit clearer, though I am hesitant to do so given current timing :)